src/main/java/org/opentripplanner/middleware/controllers/api/UserController.java

```diff
@@ -67,4 +95,44 @@ boolean preDeleteHook(User user, Request req) {
         }
         return true;
     }
+
+    /**
+     * Holds result and message from getUserFromProfile.
+     */
+    static class UserFromProfileResult {
+        public User user;
+        public String message;
+    }
+
+    /**
+     * HTTP endpoint to get the {@link User} entity from a {@link Auth0UserProfile}.
+     * (Reminder: for endpoints under 'secure', we add a {@link Auth0UserProfile} to request attributes.)
+     */
+    private User getUserFromRequest(Request req, Response res) {
+        Auth0UserProfile profile = req.attribute("user");
```

Comment:
I'm wondering if all actions related to user auth should be outside of the controllers as similar checks are likely to be done by other controllers. For instance, under the monitored-trip branch I have created: 

https://github.com/ibi-group/otp-middleware/blob/monitored-trip/src/main/java/org/opentripplanner/middleware/auth/Auth0Utils.java

Reply:
So that everyone is aware, [`Auth0Connection#checkUser`](https://github.com/ibi-group/otp-middleware/blob/94dcefa65446530ba5854afb442aac7d416c5312/src/main/java/org/opentripplanner/middleware/auth/Auth0Connection.java#L38) checks that the user is authorized. This check is run on every request method [behind a `api/secure/*` or `api/admin/*` route](https://github.com/ibi-group/otp-middleware/blob/fced81463189960895df9cfa4eee26eef9637151/src/main/java/org/opentripplanner/middleware/spark/Main.java#L79-L86).